Jenkinsfile-upgrade

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env groovy
+
+def lib = evaluate readTrusted('./jenkins-functions.groovy')
+
+pipeline {
+    agent {
+        node {
+            label 'enmasse'
+        }
+    }
+    parameters {
+        string(name: 'TEST_CASE', defaultValue: 'common.upgrade.**', description: 'maven parameter for executing specific tests')
+        string(name: 'BUILD_TAG', defaultValue: '', description: 'version of images')
+        string(name: 'COMMIT_SHA', defaultValue: '', description: 'Commit sha1')
+        string(name: 'START_VERSION', defaultValue: '0.23.1', description: 'Version of enmasse which will be upgraded')
+    }
+    options {
+        timeout(time: 1, unit: 'HOURS')
+        ansiColor('xterm')
+    }
+    environment {
+        CORES_DIR = "/tmp/cores"
+        ARTIFACTS_DIR = 'artifacts'
+        ANSIBLE_FORCE_COLOR = true
+    }
+    stages {
+        stage('wait for agent ready') {
+            steps {
+                script {
+                    lib.waitUntilAgentReady()
+                }
+            }
+        }
+        stage('add description') {
+            steps {
+                script {
+                    currentBuild.description = "related build: ${params.BUILD_TAG}"
+                }
+            }
+        }
+        stage('clean') {
+            steps {
+                cleanWs()
+            }
+        }
+        stage('checkout') {
+            steps {
+                checkout scm
+                sh 'rm -rf artifacts && mkdir -p artifacts'
+            }
+        }
+        stage('start openshift') {
+            steps {
+                timeout(time: 10, unit: 'MINUTES') {
+                    script {
+                        lib.startOpenshift()
+                    }
+                }
+            }
+        }
+        stage('deploy enmasse'){
+            environment {
+                JOB_NAME_SUB = "${String.format("%.15s", JOB_NAME)}"
+                OPENSHIFT_PROJECT = "${JOB_NAME_SUB}${BUILD_NUMBER}"
+            }
+            steps{
+                //sh "git checkout ${params.COMMIT_SHA}"
+                sh "git checkout tags/${params.START_VERSION}"
+                sh 'make templates || true'
+                sh "./systemtests/scripts/deploy_enmasse.sh false 'templates/build/enmasse-latest' true true"
```

Comment:
I'm not completely sure you will get the correct tags for the images this way. I wonder if you might need to set COMMIT=${params.START_VERSION} and TAG=${params.START_VERSION}

Reply:
you're right 👍

Reply:
and with TAG it is not true, I saw that templates are built into enmasse-latest folder. We will see after upgrade what will happen in CI

Reply:
Yeah, I will submit a PR upstream to cleanup the use of COMMIT vs. TAGS. IMO the former should not be needed anymore